packages/internal-non-workflow-common/src/otel.ts

```diff
@@ -41,7 +44,7 @@ export async function extractSpanContextFromHeaders(headers: Headers): Promise<o
 export async function headersWithContext(headers: Headers): Promise<Headers> {
   const carrier = {};
   otel.propagation.inject(otel.context.active(), carrier, otel.defaultTextMapSetter);
-  return { ...headers, [TRACE_HEADER]: toPayload(defaultPayloadConverter, carrier) };
+  return { ...headers, [TRACE_HEADER]: toPayload(payloadConverter, carrier) };
```

Comment:
Eventually I think we should just make interceptor headers `Record<string, unknown>` and apply the converter in framework code.
Not 100% sure about that though.